libchisel/src/dropsection.rs

```diff
@@ -0,0 +1,116 @@
+use super::{ModuleError, ModuleTranslator};
+
+use parity_wasm::builder::*;
+use parity_wasm::elements::*;
+
+/// Enum on which ModuleTranslator is implemented.
+pub enum DropSection<'a> {
+    NamesSection,
+    /// Name of the custom section.
+    CustomSectionByName(&'a String),
+    /// Index of the custom section.
+    CustomSectionByIndex(usize),
+    /// Index of the unknown section.
+    UnknownSectionByIndex(usize),
+}
+
+fn names_section_index_for(module: &Module) -> Result<usize, String> {
```

Comment:
These can be changed into `Option`. Or could introduce `ModuleError::NotFound`. (Actually did that in a commit, but haven't pushed.)